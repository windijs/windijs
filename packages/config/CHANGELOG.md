# @windijs/config

## 1.0.6

### Patch Changes

- [`134427d6`](https://github.com/windijs/windijs/commit/134427d666a61704de77d52f0e6e4ffbfe085cd3) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: export package.json

* [`99ecf595`](https://github.com/windijs/windijs/commit/99ecf5956d7fa1124c1561d8f58b7a0a2f1af0f2) Thanks [@voorjaar](https://github.com/voorjaar)! - build: switch to es2018 for better tree shaking

* Updated dependencies [[`134427d6`](https://github.com/windijs/windijs/commit/134427d666a61704de77d52f0e6e4ffbfe085cd3), [`99ecf595`](https://github.com/windijs/windijs/commit/99ecf5956d7fa1124c1561d8f58b7a0a2f1af0f2)]:
  - @windijs/shared@1.0.5

## 1.0.5

### Patch Changes

- [`b38b5607`](https://github.com/windijs/windijs/commit/b38b5607dccfb2d5b22af15f1b62af9ad9caf3ae) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: complete config define

* [`3bfacd6e`](https://github.com/windijs/windijs/commit/3bfacd6e745c01f650f1f5e7b65574585ee8c5bc) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: important config with string

* Updated dependencies [[`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75), [`831cae71`](https://github.com/windijs/windijs/commit/831cae71465e13a4d00188cca914710c54cd29cf)]:
  - @windijs/shared@1.0.4

## 1.0.5-beta.3

### Patch Changes

- [`3bfacd6e`](https://github.com/windijs/windijs/commit/3bfacd6e745c01f650f1f5e7b65574585ee8c5bc) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: important config with string

## 1.0.5-beta.2

### Patch Changes

- [`b38b5607`](https://github.com/windijs/windijs/commit/b38b5607dccfb2d5b22af15f1b62af9ad9caf3ae) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: complete config define

## 1.0.5-beta.1

### Patch Changes

- Updated dependencies [[`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75)]:
  - @windijs/shared@1.0.4-beta.1

## 1.0.5-beta.0

### Patch Changes

- Updated dependencies [[`831cae71`](https://github.com/windijs/windijs/commit/831cae71465e13a4d00188cca914710c54cd29cf)]:
  - @windijs/shared@1.0.4-beta.0

## 1.0.4

### Patch Changes

- af3c0c04: feat: add defineConfig interface
- Updated dependencies [939f0398]
  - @windijs/shared@1.0.3

## 1.0.3

### Patch Changes

- fix: should publish CHANGELOG.md
- Updated dependencies
  - @windijs/shared@1.0.2

## 1.0.2

### Patch Changes

- Updated dependencies
  - @windijs/shared@1.0.1

## 1.0.1

### Patch Changes

- 98ef7277: fix: package exports
