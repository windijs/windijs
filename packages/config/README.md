# @windi/config

The config variables of windi utilities. Also modular, you can import or expand the config of corresponding utility. The naming rule is `camelCased(utilityName) + Config`. For example, `backgroundSizeConfig`.

## Config

- alignContentConfig
- alignItemsConfig
- alignSelfConfig
- animationConfig
- animationDelayConfig
- animationDurationConfig
- aspectRatioConfig
- backgroundAttachmentConfig
- backgroundClipConfig
- backgroundImageConfig
- backgroundOriginConfig
- backgroundPositionConfig
- backgroundRepeatConfig
- backgroundSizeConfig
- blendModeConfig
- blurConfig
- borderRadiusConfig
- borderStyleConfig
- borderWidthConfig
- boxDecorationBreakConfig
- boxShadowConfig
- brightnessConfig
- contrastConfig
- degreeConfig
- dropShadowConfig
- flexDirectionConfig
- flexGrowConfig
- flexShrinkConfig
- flexStretchConfig
- flexWrapConfig
- fontFamilyConfig
- fontSizeConfig
- fontStyleConfig
- fontVariantNumericConfig
- fontWeightConfig
- gradientConfig
- gradientDirectionConfig
- grayscaleConfig
- gridAutoColumnsConfig
- gridAutoFlowConfig
- gridAutoRowsConfig
- gridColumnConfig
- gridColumnEndConfig
- gridColumnStartConfig
- gridRowConfig
- gridRowEndConfig
- gridRowStartConfig
- gridTemplateColumnsConfig
- gridTemplateRowsConfig
- heightConfig
- hueRotateConfig
- hyphensConfig
- imageRenderingConfig
- insetConfig
- invertConfig
- justifyContentConfig
- justifyItemsConfig
- justifySelfConfig
- letterSpacingConfig
- lineHeightConfig
- listStylePositionConfig
- listStyleTypeConfig
- marginConfig
- objectFitConfig
- objectPositionConfig
- opacityConfig
- outlineOffsetConfig
- overflowConfig
- overscrollConfig
- paddingConfig
- perspectiveConfig
- perspectiveOriginConfig
- placeContentConfig
- placeItemsConfig
- placeSelfConfig
- positionConfig
- ringOffsetConfig
- rotateConfig
- saturateConfig
- scaleConfig
- sepiaConfig
- skewConfig
- spaceBetweenConfig
- spacingConfig
- strokeLineCapConfig
- strokeLineJoinConfig
- tabSizeConfig
- tableDisplayConfig
- textAlignConfig
- textDecorationOffsetConfig
- textDecorationStyleConfig
- textDecorationThicknessConfig
- textDecorationTypeConfig
- textIndentConfig
- textShadowConfig
- textStrokeWidthConfig
- textTransformConfig
- touchActionConfig
- transformOriginConfig
- transformStyleConfig
- transitionDelayConfig
- transitionDurationConfig
- transitionPropertyConfig
- transitionTimingFunctionConfig
- translateConfig
- userSelectConfig
- verticalAlignConfig
- whiteSpaceConfig
- widthConfig
- writingModeConfig
- writingOrientationConfig
- zIndexConfig

## Keyframes

- backInUpKeyframes
- backInDownKeyframes
- backInLeftKeyframes
- backInRightKeyframes
- backOutUpKeyframes
- backOutDownKeyframes
- backOutLeftKeyframes
- backOutRightKeyframes
- backInDownKeyframes
- backInLeftKeyframes
- backInRightKeyframes
- backInUpKeyframes
- backOutDownKeyframes
- backOutLeftKeyframes
- backOutRightKeyframes
- backOutUpKeyframes
- bounceInDownKeyframes
- bounceInKeyframes
- bounceInLeftKeyframes
- bounceInRightKeyframes
- bounceInUpKeyframes
- bounceKeyframes
- bounceOutDownKeyframes
- bounceOutKeyframes
- bounceOutLeftKeyframes
- bounceOutRightKeyframes
- bounceOutUpKeyframes
- bubbleKeyframes
- fadeInBottomLeftKeyframes
- fadeInBottomRightKeyframes
- fadeInDownBigKeyframes
- fadeInDownKeyframes
- fadeInKeyframes
- fadeInLeftBigKeyframes
- fadeInLeftKeyframes
- fadeInRightBigKeyframes
- fadeInRightKeyframes
- fadeInTopLeftKeyframes
- fadeInTopRightKeyframes
- fadeInUpBigKeyframes
- fadeInUpKeyframes
- fadeOutBottomLeftKeyframes
- fadeOutBottomRightKeyframes
- fadeOutDownBigKeyframes
- fadeOutDownKeyframes
- fadeOutKeyframes
- fadeOutLeftBigKeyframes
- fadeOutLeftKeyframes
- fadeOutRightBigKeyframes
- fadeOutRightKeyframes
- fadeOutTopLeftKeyframes
- fadeOutTopRightKeyframes
- fadeOutUpBigKeyframes
- fadeOutUpKeyframes
- flashKeyframes
- flipInXKeyframes
- flipInYKeyframes
- flipKeyframes
- flipOutXKeyframes
- flipOutYKeyframes
- headShakeKeyframes
- heartBeatKeyframes
- hingeKeyframes
- jackInTheBoxKeyframes
- jelloKeyframes
- lightSpeedInLeftKeyframes
- lightSpeedInRightKeyframes
- lightSpeedOutLeftKeyframes
- lightSpeedOutRightKeyframes
- pingKeyframes
- pulseKeyframes
- rollInKeyframes
- rollOutKeyframes
- rotateInDownLeftKeyframes
- rotateInDownRightKeyframes
- rotateInKeyframes
- rotateInUpLeftKeyframes
- rotateInUpRightKeyframes
- rotateOutDownLeftKeyframes
- rotateOutDownRightKeyframes
- rotateOutKeyframes
- rotateOutUpLeftKeyframes
- rotateOutUpRightKeyframes
- rubberBandKeyframes
- shakeXKeyframes
- shakeYKeyframes
- shockKeyframes
- slideInDownKeyframes
- slideInLeftKeyframes
- slideInRightKeyframes
- slideInUpKeyframes
- slideOutDownKeyframes
- slideOutLeftKeyframes
- slideOutRightKeyframes
- slideOutUpKeyframes
- spinKeyframes
- swingKeyframes
- tadaKeyframes
- wobbleKeyframes
- zoomInDownKeyframes
- zoomInKeyframes
- zoomInLeftKeyframes
- zoomInRightKeyframes
- zoomInUpKeyframes
- zoomOutDownKeyframes
- zoomOutKeyframes
- zoomOutLeftKeyframes
- zoomOutRightKeyframes
- zoomOutUpKeyframes