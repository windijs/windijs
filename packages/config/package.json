{
  "name": "@windijs/config",
  "version": "1.0.9",
  "description": "@windijs/config",
  "main": "dist/config.js",
  "module": "dist/config.mjs",
  "types": "dist/config.d.ts",
  "files": [
    "dist",
    "CHANGELOG.md"
  ],
  "keywords": [
    "css",
    "css-in-js",
    "windi",
    "windijs",
    "windicss"
  ],
  "exports": {
    "./package.json": "./package.json",
    ".": {
      "import": "./dist/config.mjs",
      "require": "./dist/config.js"
    }
  },
  "author": "Raven Satir",
  "license": "MIT",
  "typedoc": {
    "entryPoint": "./src/index.ts",
    "displayName": "@windijs/config"
  }
}
