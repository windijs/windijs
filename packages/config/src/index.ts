export * from "./bg";
export * from "./svg";
export * from "./flex";
export * from "./grid";
export * from "./border";
export * from "./effects";
export * from "./behaviors";
export * from "./filter";
export * from "./opacity";
export * from "./sizing";
export * from "./spacing";
export * from "./animation";
export * from "./transform";
export * from "./positioning";
export * from "./interactivity";
export * from "./transitions";
export * from "./typography";
export * from "./keyframes";
export * from "./interface";
