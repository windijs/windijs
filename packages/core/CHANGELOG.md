# @windijs/core

## 1.1.0-beta.0

### Minor Changes

- [`a8e51fe2`](https://github.com/windijs/windijs/commit/a8e51fe2d193dc7e6b450fb4e7b5ad6ab330bd33) Thanks [@voorjaar](https://github.com/voorjaar)! - refactor: new handler api with meta data

### Patch Changes

- Updated dependencies [[`a8e51fe2`](https://github.com/windijs/windijs/commit/a8e51fe2d193dc7e6b450fb4e7b5ad6ab330bd33)]:
  - @windijs/helpers@1.1.0-beta.0

## 1.0.5

## 1.0.4

### Patch Changes

- Updated dependencies [939f0398]
  - @windijs/shared@1.0.3
  - @windijs/helpers@1.0.3

## 1.0.3

### Patch Changes

- fix: should publish CHANGELOG.md
- Updated dependencies
  - @windijs/helpers@1.0.2
  - @windijs/shared@1.0.2

## 1.0.2

- chore: release v1.0.2

## 1.0.1

### Patch Changes

- chore: release v1.0.1
- Updated dependencies
  - @windijs/helpers@1.0.1
  - @windijs/shared@1.0.1

## 1.0.0

### Patch Changes

- @windijs/helpers@1.0.0
