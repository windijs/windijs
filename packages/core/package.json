{
  "name": "@windijs/core",
  "version": "1.1.0-beta.2",
  "description": "@windijs/core",
  "main": "dist/core.js",
  "module": "dist/core.mjs",
  "types": "dist/core.d.ts",
  "files": [
    "dist",
    "CHANGELOG.md"
  ],
  "keywords": [
    "css",
    "css-in-js",
    "windi",
    "windijs",
    "windicss"
  ],
  "exports": {
    ".": {
      "import": "./dist/core.mjs",
      "require": "./dist/core.js"
    }
  },
  "author": "Raven Satir",
  "license": "MIT",
  "dependencies": {
    "@windijs/shared": "1.0.4-beta.0",
    "@windijs/helpers": "1.1.0-beta.2"
  },
  "typedoc": {
    "entryPoint": "./src/index.ts",
    "displayName": "@windijs/core"
  }
}
