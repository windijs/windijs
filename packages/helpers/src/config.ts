import { Handler, StyleLoader, StyleObject } from "./types";

import { Theme } from "@windijs/config";

export type DarkModeConfig = "class" | "media" | false;

export type UtilitiesConfig = { [key: string]: Handler<unknown> | Array<Handler<unknown>> | StyleObject | UtilitiesConfig };

export type VariantsConfig = { [key: string]: string | string[] };

export interface Config<T extends object = Theme> {
  // presets?: Config[];
  // prefixer?: boolean;
  // attributify?: boolean | {
  //   prefix?: string
  //   separator?: string
  //   disable?: string[]
  // }
  separator?: string;
  important?: boolean | string;
  darkMode?: DarkModeConfig;
  theme?: T;
  // variantOrder?: string[];
  // plugins?: Plugin[];
  // handlers?: Handlers;
  // corePlugins?: (keyof BaseTheme)[] | string[] | { [ T in keyof BaseTheme ] : boolean } | { [ key:string ] : boolean };
  // prefix?: string;
  // exclude?: RegExp[];
  // alias?: { [key:string]: string };
  // shortcuts?: { [key:string]: Shortcut };
  // purge?: unknown;
  utilities?: UtilitiesConfig;
  variants?: VariantsConfig;
  styleLoader?: StyleLoader;
  [key:string]: any;
}

export function defineConfig (config: Config) {
  return config;
}
