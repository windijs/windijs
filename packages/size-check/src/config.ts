import { paddingConfig } from "@windijs/config";

export { paddingConfig };
