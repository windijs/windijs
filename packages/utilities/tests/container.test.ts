import { buildContainer } from "@windi/core";
import { screens } from "@windi/config";
import { unify } from "@windi/helpers";

test("Container", () => {
  const container = buildContainer(screens);
  expect(container.css).toMatchSnapshot();
});

test("Container center", () => {
  const container = buildContainer(screens, true);
  expect(container.css).toMatchSnapshot();
});

test("Container padding", () => {
  const container = buildContainer({
    DEFAULT: ["", { padding: "1rem" }],
    sm: ["640px", { padding: "2rem" }],
    md: "768px",
    lg: ["1024px", { padding: "4rem", margin: "2rem" }],
    xl: "1280px",
    "2xl": "1536px",
  });
  expect(container.css).toMatchSnapshot();
});

test("Container compile", () => {
  const container = buildContainer(screens);
  expect(unify(".container", container)).toMatchSnapshot();
});
