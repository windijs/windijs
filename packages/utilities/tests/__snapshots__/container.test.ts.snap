// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container 1`] = `
Map {
  "width" => "100%",
  "@media (min-width: 640px)" => Object {
    "maxWidth": "640px",
  },
  "@media (min-width: 768px)" => Object {
    "maxWidth": "768px",
  },
  "@media (min-width: 1024px)" => Object {
    "maxWidth": "1024px",
  },
  "@media (min-width: 1280px)" => Object {
    "maxWidth": "1280px",
  },
  "@media (min-width: 1536px)" => Object {
    "maxWidth": "1536px",
  },
}
`;

exports[`Container center 1`] = `
Map {
  "width" => "100%",
  "marginLeft" => "auto",
  "marginRight" => "auto",
  "@media (min-width: 640px)" => Object {
    "maxWidth": "640px",
  },
  "@media (min-width: 768px)" => Object {
    "maxWidth": "768px",
  },
  "@media (min-width: 1024px)" => Object {
    "maxWidth": "1024px",
  },
  "@media (min-width: 1280px)" => Object {
    "maxWidth": "1280px",
  },
  "@media (min-width: 1536px)" => Object {
    "maxWidth": "1536px",
  },
}
`;

exports[`Container compile 1`] = `
".container {
  width: 100%;
}

@media (min-width: 640px) {
  .container {
    max-width: 640px;
  }
}

@media (min-width: 768px) {
  .container {
    max-width: 768px;
  }
}

@media (min-width: 1024px) {
  .container {
    max-width: 1024px;
  }
}

@media (min-width: 1280px) {
  .container {
    max-width: 1280px;
  }
}

@media (min-width: 1536px) {
  .container {
    max-width: 1536px;
  }
}"
`;

exports[`Container padding 1`] = `
Map {
  "width" => "100%",
  "padding" => "1rem",
  "@media (min-width: 640px)" => Object {
    "maxWidth": "640px",
    "padding": "2rem",
  },
  "@media (min-width: 768px)" => Object {
    "maxWidth": "768px",
  },
  "@media (min-width: 1024px)" => Object {
    "margin": "2rem",
    "maxWidth": "1024px",
    "padding": "4rem",
  },
  "@media (min-width: 1280px)" => Object {
    "maxWidth": "1280px",
  },
  "@media (min-width: 1536px)" => Object {
    "maxWidth": "1536px",
  },
}
`;
